tests/ComponentLocaleIsPersistedTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Tests;
+
+use Illuminate\Support\Facades\App;
+use Livewire\Livewire;
+use Livewire\Component;
+
+class ComponentLocaleIsPersisted extends TestCase
+{
+    /** @test */
+    public function a_livewire_component_can_persist_its_locale()
+    {
```

Comment:
This test is a tad confusing to me.

Here's how I would re-structure:

1. Set the app locale
2. Boot up the component and check that the locale is set properly from mount to match the previously set locale
3. Change the local
4. Run a component method called "refreshLocale" (instead of "check")
5. Check that the right locale is set